components/admin_console/team_channel_settings/channel/details/channel_moderation.tsx

```diff
@@ -0,0 +1,349 @@
+            defaultMessage: 'Members can not use channel mentions without the ability to create posts.'
+        },
+        disabledBothDueToCreatePosts: {
+            id: t('admin.channel_settings.channel_moderation.channelMentions.disabledBothDueToCreatePosts'),
+            defaultMessage: 'Guests and members can not use channel mentions without the ability to create posts.'
+        }
+    },
+
+    title: {
+        id: t('admin.channel_settings.channel_moderation.title'),
+        defaultMessage: 'Channel Moderation',
+    },
+    subtitle: {
+        id: t('admin.channel_settings.channel_moderation.subtitle'),
+        defaultMessage: 'Manage the actions available to channel members and guests.'
+    },
+    permissions: {
+        id: t('admin.channel_settings.channel_moderation.permissions'),
+        defaultMessage: 'Permissions'
+    },
+    guests: {
+        id: t('admin.channel_settings.channel_moderation.guests'),
+        defaultMessage: 'Guests',
+    },
+    members: {
+        id: t('admin.channel_settings.channel_moderation.members'),
+        defaultMessage: 'Members',
+    }
+});
+
+interface Props {
+    channelPermissions?: Array<ChannelPermissions>;
+    onChannelPermissionsChanged: (name: string, channelRole: 'guests' | 'members') => void;
+    teamSchemeID?: string;
+    teamSchemeDisplayName?: string;
+}
+
+interface RowProps {
+    name: string;
+    guests?: boolean;
+    members: boolean;
+    guestsDisabled?: boolean;
+    membersDisabled: boolean;
+    onClick: (name: string, channelRole: 'guests' | 'members') => void;
+    errorMessages?: any;
+}
+
+const ChannelModerationTableRow: React.FunctionComponent<RowProps> = (props: RowProps): JSX.Element => {
+    return (
+        <tr>
+            <td>
+                <label>
+                    <FormattedMessage
+                        id={formattedMessages[props.name].title.id}
+                        defaultMessage={formattedMessages[props.name].title.defaultMessage}
+                    />
+                </label>
+                <div>
+                    <FormattedMessage
+                        id={formattedMessages[props.name].description.id}
+                        defaultMessage={formattedMessages[props.name].description.defaultMessage}
+                    />
+                </div>
+                {props.errorMessages}
+            </td>
+            <td>
+                {!isNil(props.guests) &&
+                    <button
+                        data-testid={`${props.name}-guests`}
+                        className={classNames(
+                            'checkbox',
+                            {
+                                checked: props.guests && !props.guestsDisabled,
+                                disabled: props.guestsDisabled,
+                            }
+                        )}
+                        onClick={() => props.onClick(props.name, 'guests')}
```

Comment:
Maybe @hmhealey can confirm but I *think* these inline function definition cause unnecessary re-renders.

Reply:
I tried removing the `()` but I got this error which is why I did it that way. Thoughts on how to avoid inline function without getting the error? 
![image](https://user-images.githubusercontent.com/17804942/75584743-6b20b800-5a3e-11ea-8c09-d2b44896b872.png)